src/webgpu/util/conversion.ts

```diff
@@ -991,12 +1034,30 @@ export interface ScalarBuilder {
 function scalarFromValue(
   type: ScalarType,
   workingDataArray: TypedArrayBufferView,
-  value: number
+  value: number | bigint
```

Comment:
Can't you use the `ArrayElementType<A>` trick above to statically enforce the correct value type?

Reply:
Nice :+1: